docs/web/docs/explanations/architect_api.md

```diff
@@ -0,0 +1,108 @@
+1. Connecting
+2. Onboarding
+3. Waiting (for other `Unit`s in an assignment)
+4. In-Task
+5. Completed (task complete, or partner disconnects)
+6. Disconnected (Server disconnect, timeout, return)
+7. Failed to connect (no available tasks)
+
+We've also considered a "post-task" state after the completion of a task for surveys or related content.
+
+The routing server is responsible for keeping track of the liveliness of individual `Agent`s. If it observes a disconnect on the socket, as well as timeouts on heartbeat packets.
+
+Certain status transitions will come in from the main server, and the router may be responsible for cleaning up local state or caching results at this stage.
+
+### RESTful vs Socket interactions
+We've divided Mephisto tasks into two primary types, `static` and `live` tasks. The former shouldn't require backend access through the majority of the task, only during key points (starting, submission), while the latter can have direct communication throughout the task. 
+
+Beyond just being simpler to implement, `static` tasks also have the advantage of being lenient on worker behavior; if a worker suspends progress and returns within the timeout window, they aren't penalized, even if their machine were to sleep during that window.
+
+The Mephisto backend channels expect to communicate with the router in a certain way. Our primary `Channel` is the `WebsocketChannel`, and as such we expect to receive `Packet`s over the wire from the routing server.
+
+## Implementation (proposed)
+
+The core carrier for information in the Mephisto Architect API is the `Packet` class. Downstream, they act as the way to tie an `Agent` class in python to an actual human worker. 
+
+The `Channel` is the primary way of trasmitting packets, with the `WebsocketChannel` being the main implementation Mephisto currently uses with its `Architect`s. The `ClientIOHandler` is responsible for using and interpreting packets, so it defines the key types to be handled.
+
+### Packet Types
+There are a number of different packet types used by Mephisto:
+- `PACKET_TYPE_ALIVE`: Used to mark the success of a new socket connection.
+- `PACKET_TYPE_SUBMIT`: Used to handle submission of any subcomponent of a `Unit`, currently either onboarding or full task. 
+- `PACKET_TYPE_OUTGOING_LIVE_DATA`: Used to send any new live data to an agent.
+- `PACKET_TYPE_INCOMING_LIVE_DATA`: Used for the frontend to send any type of data to the backend, usually to be processed by a user-defined callback. 
+- `PACKET_TYPE_REGISTER_WORKER`: Used to request a new `Agent` and `Unit` data for a specific worker on a task.
+- `PACKET_TYPE_AGENT_DETAILS`: Used to respond with the details of a worker registration request.
+- `PACKET_TYPE_REQUEST_STATUSES`: Used by Mephisto to poll for the current statuses for a worker.
+- `PACKET_TYPE_RETURN_STATUES`: Used by the router to return updates for all of the currently registered agents.
```

Comment:
[Nit]

Says statues instead of Statuses

Reply:
Noted and fixed, thanks!